File: CSharp/k06_1.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace CSharp
{
    public class K06_1 : IExecutable
    {
        public void Execute()
        {
            List<int> ages = new List<int>();

            for (int i = 0; i < 5; i++)
            {
                Console.Write((i + 1) + "人目の年齢を入力して下さい：");
                ages.Add(int.Parse(Console.ReadLine()));
            }

            Console.WriteLine(string.Format("{0}人の平均年齢は{1}です。", ages.Count, ages.Average()));
        }
    }
}

File: Python/k06_1.py
from functools import reduce

class k06_1(object):
    
    def execute(self):
        try:
            ages = []
            
            for i in range(0, 5):
                print('{}人目の年齢を入力して下さい：'.format(i + 1))
                age = int(input())
                ages.append(age)

            average = reduce(lambda x, y: x + y, ages) / len(ages)
            
            print('{}人の平均年齢は{}です。'.format(len(ages), average))

        except ValueError as ve:
            print(ve)